src/controller/tests/data_model/TestRead.cpp

```diff
@@ -1672,12 +1698,19 @@ void TestReadInteraction::TestReadHandler_KillOverQuotaSubscriptions(nlTestSuite
 
     // The following check will trigger the logic in im to kill the read handlers that use more paths than the limit per fabric.
     {
-        EstablishSubscriptions(apSuite, ctx.GetSessionAliceToBob(),
-                               app::InteractionModelEngine::kMinSupportedSubscriptionsPerFabric,
-                               app::InteractionModelEngine::kMinSupportedPathsPerSubscription, &readCallbackFabric2, readClients);
-
-        ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(60),
-                                        [&]() { return readCallbackFabric2.mOnSubscriptionEstablishedCount == 1; });
+        EstablishReadOrSubscriptions(
```

Comment:
There is very little comments in this test to provide a way for readers to understand what's happening. It's taking me a while to look at the code to figure out the rationale, would be so much better if we could have good comments. 

This test has to read like a story so that readers understand the flow of how the test is progresing.

You should provide a statement about the current resource situation, what you're going to do, and what the result should look like. 

("we currently have X subscriptions established from Fabric A, and we're going to now add Y subscriptions from Fabric B. This should kick out Z subscriptions from Fabric A, etc.").